### PR TITLE
Added support for managed disk diff

### PR DESCRIPTION
**Pull request by gapra-msft**
This PR is pending a discussion about how this feature should be exposed to the customer

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java [hunk 1]
```diff
@@ -596,23 +596,88 @@ private static String pageRangeToString(PageRange pageRange) {
...
+        String prevSnapshotUrl, BlobRequestConditions requestConditions, Context context) {
         blobRange = blobRange == null ? new BlobRange(0) : blobRange;
         requestConditions = requestConditions == null ? new BlobRequestConditions() : requestConditions;
 
-        if (prevSnapshot == null) {
-            throw logger.logExceptionAsError(new IllegalArgumentException("prevSnapshot cannot be null"));
+        URL url = null;
+        if (prevSnapshotUrl == null) {
+            if (prevSnapshot == null) {
+                throw logger.logExceptionAsError(new IllegalArgumentException("prevSnapshot cannot be null"));
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java]: I think you mean both can't be null

Comment (rickle-msft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java]: They also can't both be specified, right?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java [hunk 2]
```diff
@@ -596,23 +596,88 @@ private static String pageRangeToString(PageRange pageRange) {
...
+     *
+     * {@codesnippet com.azure.storage.blob.specialized.PageBlobAsyncClient.getManagedDiskRangesDiff#BlobRange-String}
+     *
+     * @param blobRange {@link BlobRange}
+     * @param prevSnapshotUrl Specifies the URL of a previous snapshot of the target blob. Specifies that the
+     * response will contain only pages that were changed between target blob and previous snapshot. Changed pages
+     * include both updated and cleared pages. The target blob may be a snapshot, as long as the snapshot specified by
+     * prevsnapshot is the older of the two.
+     *
+     * @return A reactive response emitting all the different page ranges.
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java]: This sounds kind of weird. Can you say "all the page ranges that differ between the specified versions of the blob"?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java (hunk 1 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java]: I think this was from when we planned on adding an overload - it isnt required anymore, I'll remove it in a new PR

**Pull request merged**